java8/Runner.java

```diff
@@ -0,0 +1,41 @@
+package by.epam.lab;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+public class Runner {
+    public static void main(String[] args) {
+        ArrayList<Trial> trials = new ArrayList<Trial>(Arrays.asList(
+                new Trial("Sakovich", 45, 93),
+                new Trial("Sergeev", 55, 50),
+                new Trial("Stepanenko", 60, 90),
+                new LiteTrial("Sergeev", 29, 95),
+                new LiteTrial("Lis", 35, 68),
+                new StrongTrial("Sakovich", 90, 90),
+                new StrongTrial("Sergeev", 70, 57),
+                new ExtraTrial("Stepanenko", 40, 50, 60),
+                new ExtraTrial("Sakovich", 70, 51, 80)
+        ));
+
+        trials.forEach(System.out::println);
+
+        System.out.println("\nNumber of passed trials:" +
+                trials.stream().filter(trial -> trial.isPassed()).count());
+
+        Collections.sort(trials);
+
+        System.out.println("\nSum of marks from sorted array:");
+        trials.stream().map(Trial::getSum).forEach(System.out::println);
+
+        ArrayList<Trial> unpassed = new ArrayList<Trial>(trials.stream().filter(trial -> !trial.isPassed()).toList());
+        unpassed.forEach(Trial::clearMarks);

```

Comment:
You can do this with .peek(...) in previous stream. And... this isn't working properly. You need to clone object, in other case condition of the object change in original array. Use Clonable interface